File: longJump/src/main/kotlin/longjump.java
class Solution {
    public long solution(int n) {
        long answer = 0;
        long[] A = new long[2001];
        A[0] = 1; A[1] = 1;
        for(int i = 2; i <= n; i++)
        {
            A[i] = (A[i-1] + A[i-2]) % 1234567;
        }
        answer = A[n];
        return answer;
    }
}

File: longJump/src/main/kotlin/longJump.kt
class Solution {
    fun solution(n: Int): Long {
        var dp = IntArray(n+1)
        dp[1] = 1
        if(n>=2) dp[2] = 2
        for(i in 3..n){
            dp[i] = (dp[i-1]+dp[i-2]) % 1234567
        }
        return dp[n].toLong()
    }
} //동적 계획법 문제